src/metadata/use-metadata.js

```diff
@@ -1,34 +1,31 @@
 import { useQuery } from 'react-query'
+import { createSelector } from 'reselect'
 import { hashArraysInObject } from './utils.js'
 
-export const useMetadata = () => {
-    const queryKey = [
-        'metadata',
-        {
-            params: {
-                // Note: on dataSet.dataSetElement, the categoryCombo property is
-                // included because it can mean it's overriding the data element's
-                // native categoryCombo. It can sometimes be absent from the data
-                // set element
-                'dataSets:fields':
-                    'id,displayFormName,formType,dataSetElements[dataElement,categoryCombo],categoryCombo,sections~pluck',
-                'dataElements:fields':
-                    'id,displayFormName,categoryCombo,valueType,optionSetValue,optionSet',
-                'optionSets:fields': 'id,options[id,name,sortOrder]',
-                'sections:fields':
-                    'id,displayName,sortOrder,showRowTotals,showColumnTotals,disableDataElementAutoGroup,greyedFields[id],categoryCombos~pluck,dataElements~pluck,indicators~pluck',
-                'categoryCombos:fields':
-                    'id,skipTotal,categories~pluck,categoryOptionCombos~pluck,isDefault',
-                'categories:fields': 'id,displayFormName,categoryOptions~pluck',
-                'categoryOptions:fields':
-                    'id,displayFormName,categoryOptionCombos~pluck,categoryOptionGroups~pluck,isDefault',
-                'categoryOptionCombos:fields':
-                    'id,categoryOptions~pluck,categoryCombo,name',
-            },
-        },
-    ]
+const selectorFunction = createSelector(
+    (data) => data,
+    (data) => {
+        // bug in API includes duplicate "default"-catCombo
+        // one has categoryOptionCombos, and the other is missing properties
+        // delete the one that is missing properties, so we ensure that the correct one is used
+        const filtered = data.categoryCombos.filter(
+            (cc) => !(cc.isDefault && !cc.categoryOptionCombos)
+        )
+        const filteredData = { ...data, categoryCombos: filtered }
+        const result = hashArraysInObject(filteredData)
+        console.log(result)
```

Comment:
I suppose @Birkbjo is not done cleaning this up yet, just adding this comment so I'll remember it